contexts/DelegationContext.tsx

```diff
@@ -0,0 +1,292 @@
+import { zeroAddress } from "helpers";
+import {
+  useAcceptReceivedRequestToBeDelegate,
+  useCancelSentRequestToBeDelegate,
+  useContractsContext,
+  useDelegateToStaker,
+  useDelegatorSetEventsForDelegate,
+  useDelegatorSetEventsForDelegator,
+  useIgnoredRequestToBeDelegateAddresses,
+  useIgnoreReceivedRequestToBeDelegate,
+  usePanelContext,
+  useReceivedRequestsToBeDelegate,
+  useSendRequestToBeDelegate,
+  useSentRequestsToBeDelegate,
+  useStakingContext,
+  useTerminateRelationshipWithDelegate,
+  useTerminateRelationshipWithDelegator,
+  useUserContext,
+  useVoterFromDelegate,
+} from "hooks";
+import { createContext, ReactNode } from "react";
+import { DelegationEventT, DelegationStatusT } from "types";
+export interface DelegationContextState {
+  getDelegationStatus: () => DelegationStatusT;
+  getPendingReceivedRequestsToBeDelegate: () => DelegationEventT[];
```

Comment:
These methods define the actions that the user can take from the frontend. I have deliberately limited these actions so that initiating a delegator-delegate relationship can only be done in one direction — one can send a request to another wallet to be their delegate, but not the other way around. This greatly reduces the number of states we need to account for. Also in my opinion this will be much more comprehensible to the user.